ReverseTicTacToe/UI/ConsoleUI.cs

```diff
@@ -33,32 +31,216 @@ public  void Init()
+            }
+            else
             {
 
             }
         }
 
-        private void getBoardSizeFromUser()
+        private string getEndGameMsg(int i_RetStatus)
         {
-            string userInputStr;
+            StringBuilder stringBuilder = new StringBuilder();
+            if(i_RetStatus == Game.StatusWin)
+            {
+                //write who wins
+            }
+            else
+            {
+                stringBuilder.Append("This is a Tie!");
+            }
+           // stringBuilder.AppendFormat("Score is: {0} Player1: {1}", Environment.NewLine, m_Game
+            return stringBuilder.ToString();
+        }
+
+        private void getRowColFromUser(out int i_Row, out int i_Col)
+        {
+            string turnMsg = GetTurnMessage();
+            string userTurnInput;
+            do
+            {
+                WriteLine(turnMsg);
+                userTurnInput = ReadLine();
+            }
+            while (!turnInputIsValid(userTurnInput, out i_Row, out i_Col));
+        }
+
+        private bool turnInputIsValid(string i_UserTurnInput, out int o_Row, out int o_Col)
+        {
+            o_Row = QUIT;
+            o_Col = QUIT;
+            const bool v_IsValid = true;
+            bool turnInputIsValid = !v_IsValid;
+            i_UserTurnInput = removeWhitSpaces(i_UserTurnInput);
+            if (i_UserTurnInput == QUITSYMBOL)
+            {
+                turnInputIsValid = v_IsValid;
+            }
+            // Expecting here to have the following string: 
+            // "i,j" - where i is row and j is col.
+            else if (i_UserTurnInput.Length == 3 && i_UserTurnInput[1] == ',')
+            {
+                char[] charArray = i_UserTurnInput.ToCharArray();
+                if(char.IsDigit(charArray[0]) && char.IsDigit(charArray[2]) && charArray[1] == ',')
+                {
+                    int row = int.Parse(charArray[0].ToString());
+                    int col = int.Parse(charArray[2].ToString());
+                    //checking that the row and col are within range of board size
+                    if(isInRage(row, 0,m_Game.BoardSize) &&
+                        (isInRage(col, 0,m_Game.BoardSize)))
+                    {
+                        turnInputIsValid = v_IsValid;
+                        o_Row = row;
+                        o_Col = col;
+                    }
+                }
+            }
+            return turnInputIsValid;
+        }
+
+        private string removeWhitSpaces(string i_UserTurnInput)
+        {
+            StringBuilder stringBuilder = new StringBuilder(i_UserTurnInput);
+            int index = stringBuilder.ToString().IndexOf(' ');
+            while(index != -1)
+            {
+                stringBuilder.Remove(index, 1);
+                index = stringBuilder.ToString().IndexOf(' ');
+            }
+            return stringBuilder.ToString();
+        }
+
+        private bool isInRage(int i_Num, int i_Min, int i_Max)
```

Comment:
maybe make this a static method ?

Reply:
Are you sure?
is this method something that others will use from ConsoleUI class?
maybe we should create a Utility class for all those generic and logic methods?

Reply:
I thought about that too, but i don't think he talked about classes like these in class, did he?
I'm not sure how you make one